### PR TITLE
Create server-library-reference-specification.md

### PR DESCRIPTION
**Pull request by danielrbrowne**
- Adds v1 of the reference specification for a Channels server library.

### CODE REVIEW

File: docs/channels/library_auth_reference/server-library-reference-specification.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  - Examples of a suitable error reporting mechanism could be throwing catchable exceptions, returning errors using a delegate pattern, or logging errors to a console.
+- All HTTP API error responses SHOULD be surfaced via the library unless there is a specific reason that a HTTP API error MUST NOT be surfaced for security reasons.
+
+## Tests and code coverage
+
+The library MUST include a standardised set of unit tests, covering its critical functionality. The exact implementation SHOULD be dictated based on development language best practices.
+
+### Triggering events
+
+- `testPostBatchEventsToChannelFailsForMultichannelEvents`
```
Comment (ccaseypusher) [inline docs/channels/library_auth_reference/server-library-reference-specification.md:173]: Should there be more details for what each test is and expects?

Comment (danielrbrowne) [reply docs/channels/library_auth_reference/server-library-reference-specification.md:173]: I _could_ add some additional details I suppose, however I didn't want to be overly prescriptive and I thought the test names (which are taken from the Channels Swift HTTP SDK) described the high-level behaviour of each test case. What do you think?

**Review by GeorgeRDBuckley — APPROVED** [review]
✌️ LGTM

File: docs/channels/library_auth_reference/server-library-reference-specification.md (hunk 1 above)
Comment (ccaseypusher) [reply docs/channels/library_auth_reference/server-library-reference-specification.md:173]: Perhaps a note along the lines of "the tests are self descriptive, but examples can be found in pusher-http-swift".

Comment (danielrbrowne) [reply docs/channels/library_auth_reference/server-library-reference-specification.md:173]: This is a good compromise - I'll push a change now and then get this merged in :)

**Pull request merged**